api/lib/gateways/Academy-Benefits/Search.js

```diff
@@ -1,80 +1,76 @@
-const path = require('path');
-const { loadSQL } = require('../../Utils');
-const { Systems } = require('../../Constants');
-const { searchCustomersSQL } = loadSQL(path.join(__dirname, 'sql'));
+const moment = require("moment");
 
 module.exports = options => {
-  const db = options.db;
-  const buildSearchRecord = options.buildSearchRecord;
+  const searchDb = options.searchDb;
+  const searchAPI = options.searchAPI;
   const logger = options.logger;
 
-  const search = async queryParams => {
-    let whereClause = [];
-    let params = [];
-    if (queryParams.firstName && queryParams.firstName !== '') {
-      params.push({
-        id: 'forename',
-        type: 'NVarChar',
-        value: `%${queryParams.firstName.toUpperCase().trim()}%`
-      });
-      whereClause.push('forename LIKE @forename');
+  const getDbRecords = async queryParams => {
+    try {
+      return await searchDb.execute(queryParams);
+    } catch (err) {
+      logger.error(
+        `Error searching customers in Academy-Benefits: ${err}`,
+        err
+      );
+      return [];
     }
-    if (queryParams.lastName && queryParams.lastName !== '') {
-      params.push({
-        id: 'surname',
-        type: 'NVarChar',
-        value: `%${queryParams.lastName.toUpperCase().trim()}%`
-      });
-      whereClause.push('surname LIKE @surname');
+  }
+
+  const getApiRecords = async queryParams => {
+    try {
+      return await searchAPI.execute(queryParams);
+    } catch (err) {
+      logger.error(
+        `Error searching customers in Academy-Benefits API: ${err}`,
+        err
+      );
+      return [];
     }
+  }
 
-    whereClause = whereClause.map(clause => `(${clause})`);
-    const query = `${searchCustomersSQL} AND(${whereClause.join(' AND ')})`;
-    return await db.request(query, params);
-  };
+  const checkRecordsAreIdentical = (dbRecords, apiRecords) => {
+    if (sameResults(dbRecords, apiRecords)) {
+      logger.log("Academy records retrieved from the API and the DB are identical");
+    } else {
+      logger.log("Academy API and DB have returned different records");
+      logger.log({ "DB records": dbRecords })
+      logger.log({ "API records": apiRecords })
+    }
+  }
 
-  const validateIds = record => {
-    return record.claim_id && record.check_digit && record.person_ref;
-  };
+  const sameResults = (dbRecords, apiRecords) => {
+    if (dbRecords.length !== apiRecords.length) return false;
+    let equal = true;
+    dbRecords.forEach(record => {
+      if (!apiRecords.find(r => recordEquality(record, r))) {
+        equal = false;
```

Comment:
can equal be set to true again here?

Reply:
I'm not sure I know what you mean. 
The aim is for it to go through each record returned from the database and check that it has an equivalent record from the API and if not to set equal to false. If every record is found the API records, the equal variable will remain true. - Does that make sense?

Reply:
Yeah so if the first record is not is apiRecords, equal is set to false. But then if the second record is apiRecords, equal will get set to true again? Or am I reading it wrong? I think it maybe needs a break?

Reply:
It won't reset it to true, it'll just do nothing. But equally a break does make sense, no point continuing the loop if it already failed so I'll add it in

Reply:
Haha, never mind me!